src/notification/notification.controller.ts

```diff
@@ -13,7 +15,8 @@ export class NotificationController {
 
     @Post('saveNotificationToDb')
     async saveNotificationToDb(@Body() dto: SaveNotificationDto){
-        return await this.notificationService.saveNotificationToDb(dto)
+        const savedNotification = await this.notificationService.saveNotificationToDb(dto)
```

Comment:
do we really need this saveNotification to be an endpoint?

And how will other BE services connect with this if they want to notify a user for something

Reply:
I left the endpoint there so the function could easily be tested by reviewers.

The function can be called by the functions that are emitting the websocket events to notify users. That way the notifications are saved immediately after the events are fired.